OBAKit/Helpers/OBAStopIconFactory.m

```diff
@@ -16,38 +16,298 @@
+                                    mainViewRect.size.width * ChevronWidthPercentage, mainViewRect.size.height * ChevronHeightPercentage);
+
+    // Glyph overall size is governed by the height of the glyph area
+    CGFloat glyphHeight = iconRect.size.height * GlyphHeightPercentage;
+    CGFloat glyphWidth = iconRect.size.width - (strokeWidth * 2);
+    CGFloat glyphX = (iconRect.origin.x + (iconRect.size.width / 2)) - (glyphWidth / 2);
+    CGFloat glyphY = iconRect.origin.y + (strokeWidth * GlyphInsetRelativeToStroke);
+    CGRect glyphRect = CGRectMake(glyphX, glyphY, glyphWidth, glyphHeight);
+
+    // The first thing to do is create a view that will hold everything
+    UIView *view = [[UIView alloc] initWithFrame:mainViewRect];
+    view.backgroundColor = [UIColor clearColor];
+
+    // Add rounded rect for the icon frame
+    OBACanvasView *rectView = [[OBACanvasView alloc] initWithFrame:iconRect drawRectBlock:^void (CGRect rect) {
+        // Inset the rectangle by half of the stroke width so that the stroke fits
+        CGRect strokeRect = CGRectInset(rect, strokeWidth / 2, strokeWidth / 2);
+
+        UIBezierPath *rectanglePath = [UIBezierPath bezierPathWithRoundedRect:strokeRect cornerRadius:cornerRadius];
 
+        [iconBackgroundColor setFill];
+        [rectanglePath fill];
+        [strokeColor setStroke];
+        rectanglePath.lineWidth = strokeWidth;
+        [rectanglePath stroke];
+    }];
+    [view addSubview:rectView];
+
+    // Add the text panel at the bottom of the icon
+    OBACanvasView *textView = [[OBACanvasView alloc] initWithFrame:textRect drawRectBlock:^void (CGRect rect) {
+        // Inset the rectangle by half of the stroke width so that the stroke fits
+        CGRect blockRect = CGRectInset(rect, strokeWidth / 2, strokeWidth / 2);
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wassign-enum"
+        // Hate this pragma, but Cocoa doesn't define 'full' or 'empty' values in its
+        // bitfield enums, which causes a warning for a value that's out of range
+        // even when it's being used exactly as intended.
+        UIBezierPath *rectanglePath = [UIBezierPath bezierPathWithRoundedRect:blockRect
+                                                            byRoundingCorners:(UIRectCornerBottomLeft | UIRectCornerBottomRight)
+                                                                  cornerRadii:CGSizeMake(cornerRadius, cornerRadius)];
+#pragma clang diagnostic pop
+
+        [textBackgroundColor setFill];
+        [rectanglePath fill];
+
+        CGRect labelRect = CGRectInset(blockRect, strokeWidth, strokeWidth);
+        CGFloat fontSize = [self findFontSizeForFrame:labelRect forFace:stopLabelFaceName withText:stopLabelText];
+
+        NSDictionary *attributes = @{ NSFontAttributeName:[UIFont fontWithName:stopLabelFaceName size:fontSize],
```

Comment:
I feel like there are diminishing returns on caching these, since the enclosing referrer is cache-optimized.